packages/jupyter-ai/jupyter_ai/personas/directories.py

```diff
@@ -0,0 +1,98 @@
+from pathlib import Path
+from typing import Optional
+
+
+def find_dot_dir(
+    dir: str, dot_dir: str, root_dir: Optional[str] = None
+) -> Optional[str]:
+    """
+    Find the nearest dot directory by traversing up from the given directory.
+
+    Args:
+        dir (str): The starting directory path
```

Comment:
It's not clear whether this path is relative to Jupyter root? Would help to add that info to the doc.